internal/trie/node/branch_encode.go

```diff
@@ -133,7 +133,7 @@ func encodeChild(child *Node, buffer io.Writer) (err error) {
 // and then SCALE encodes it. This is used to encode children
 // nodes of branches.
 func scaleEncodeHash(node *Node) (encoding []byte, err error) {
-	_, merkleValue, err := node.EncodeAndHash()
+	merkleValue, err := node.CalculateMerkleValue()
```

Comment:
`CalculateMerkleValue` uses the cached node Merkle value if it's already computed, and doesn't require to re-encode all the descendant nodes.

Since before #2919 we were caching encoding fields, we did not see the impact of this since hashing the cached encoding was cheap. This is now performance critical to use the cached merkle value to avoid re-encoding+hashing all the children.